Postgresql metrics receiver has runtime error on type conversion

### What happened?

## Description

Postgresql metrics receiver has runtime error on converting null to int64 for `write_lag` 
and `checkpoint_duration_write` float64 to int64

## Steps to Reproduce

## Expected Result

## Actual Result


### Collector version

v0.62.1

### Environment information

## Environment
OS: "Ubuntu 20.04"
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

```yaml
extensions:
receivers:
    postgresql:
            endpoint: localhost:5432
            transport: tcp
            username: postgres
            password: password
            databases:
                - test_db
            collection_interval: 10s
            tls:
                insecure: true
exporters:
    kafka:
        brokers: [ localhost:9092 ]
        protocol_version: "3.0.0"
processors:
service:
    pipelines:
        metrics:
            receivers: [ postgresql ]
            processors: []
            exporters: [ kafka ]
```


### Log output

```shell
Oct 17 07:17:31 Test-Host otelcol-contrib[3184939]: 2022-10-17T07:17:31.883Z        error        scraperhelper/scrapercontroller.go:197        Error scraping metrics        {"kind": "receiver", "name": "postgresql", "pipeline": "metrics", "error": "sql: Scan error on column index 2, name \"checkpoint_duration_write\": converting driver.Value type float64 (\"4.0602716e+08\") to a int64: invalid syntax; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported; sql: Scan error on column index 2, name \"write_lag\": converting NULL to int64 is unsupported", "scraper": "postgresql"}
```


### Additional context

_No response_

--- a/.chloggen/postgresql-wal-lag-null.yaml
+++ b/.chloggen/postgresql-wal-lag-null.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: postgresqlreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix issue where WAL lag stats could cause panic
+
+# One or more tracking issues related to the change
+issues: [14972]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/receiver/postgresqlreceiver/client.go
+++ b/receiver/postgresqlreceiver/client.go
@@ -288,15 +288,15 @@ type tableIOStats struct {
 }
 
 func (c *postgreSQLClient) getBlocksReadByTable(ctx context.Context, db string) (map[tableIdentifier]tableIOStats, error) {
-	query := `SELECT schemaname || '.' || relname AS table, 
-	coalesce(heap_blks_read, 0) AS heap_read, 
-	coalesce(heap_blks_hit, 0) AS heap_hit, 
-	coalesce(idx_blks_read, 0) AS idx_read, 
-	coalesce(idx_blks_hit, 0) AS idx_hit, 
-	coalesce(toast_blks_read, 0) AS toast_read, 
-	coalesce(toast_blks_hit, 0) AS toast_hit, 
-	coalesce(tidx_blks_read, 0) AS tidx_read, 
-	coalesce(tidx_blks_hit, 0) AS tidx_hit 
+	query := `SELECT schemaname || '.' || relname AS table,
+	coalesce(heap_blks_read, 0) AS heap_read,
+	coalesce(heap_blks_hit, 0) AS heap_hit,
+	coalesce(idx_blks_read, 0) AS idx_read,
+	coalesce(idx_blks_hit, 0) AS idx_hit,
+	coalesce(toast_blks_read, 0) AS toast_read,
+	coalesce(toast_blks_hit, 0) AS toast_hit,
+	coalesce(tidx_blks_read, 0) AS tidx_read,
+	coalesce(tidx_blks_hit, 0) AS tidx_hit
 	FROM pg_statio_user_tables;`
 
 	tios := map[tableIdentifier]tableIOStats{}
@@ -388,7 +388,7 @@ type bgStat struct {
 }
 
 func (c *postgreSQLClient) getBGWriterStats(ctx context.Context) (*bgStat, error) {
-	query := `SELECT 
+	query := `SELECT
 	checkpoints_req AS checkpoint_req,
 	checkpoints_timed AS checkpoint_scheduled,
 	checkpoint_write_time AS checkpoint_duration_write,
@@ -455,12 +455,12 @@ type replicationStats struct {
 }
 
 func (c *postgreSQLClient) getReplicationStats(ctx context.Context) ([]replicationStats, error) {
-	query := `SELECT 
+	query := `SELECT
 	client_addr,
-	coalesce(pg_wal_lsn_diff(pg_current_wal_lsn(), replay_lsn), 0) AS replication_bytes_pending,
-	write_lag,
-	flush_lag,
-	replay_lag
+	coalesce(pg_wal_lsn_diff(pg_current_wal_lsn(), replay_lsn), -1) AS replication_bytes_pending,
+	coalesce(write_lag, -1),
+	coalesce(flush_lag, -1),
+	coalesce(replay_lag, -1)
 	FROM pg_stat_replication;
 	`
 	rows, err := c.client.QueryContext(ctx, query)

--- a/receiver/postgresqlreceiver/scraper.go
+++ b/receiver/postgresqlreceiver/scraper.go
@@ -350,10 +350,18 @@ func (p *postgreSQLScraper) collectReplicationStats(
 		return
 	}
 	for _, rs := range rss {
-		p.mb.RecordPostgresqlReplicationDataDelayDataPoint(now, rs.pendingBytes, rs.clientAddr)
-		p.mb.RecordPostgresqlWalLagDataPoint(now, rs.writeLag, metadata.AttributeWalOperationLagWrite, rs.clientAddr)
-		p.mb.RecordPostgresqlWalLagDataPoint(now, rs.replayLag, metadata.AttributeWalOperationLagReplay, rs.clientAddr)
-		p.mb.RecordPostgresqlWalLagDataPoint(now, rs.flushLag, metadata.AttributeWalOperationLagFlush, rs.clientAddr)
+		if rs.pendingBytes >= 0 {
+			p.mb.RecordPostgresqlReplicationDataDelayDataPoint(now, rs.pendingBytes, rs.clientAddr)
+		}
+		if rs.writeLag >= 0 {
+			p.mb.RecordPostgresqlWalLagDataPoint(now, rs.writeLag, metadata.AttributeWalOperationLagWrite, rs.clientAddr)
+		}
+		if rs.replayLag >= 0 {
+			p.mb.RecordPostgresqlWalLagDataPoint(now, rs.replayLag, metadata.AttributeWalOperationLagReplay, rs.clientAddr)
+		}
+		if rs.flushLag >= 0 {
+			p.mb.RecordPostgresqlWalLagDataPoint(now, rs.flushLag, metadata.AttributeWalOperationLagFlush, rs.clientAddr)
+		}
 	}
 }
 

--- a/receiver/postgresqlreceiver/scraper_test.go
+++ b/receiver/postgresqlreceiver/scraper_test.go
@@ -263,6 +263,13 @@ func (m *mockClient) initMocks(database string, databases []string, index int) {
 				replayLag:    700,
 				writeLag:     800,
 			},
+			{
+				clientAddr:   "nulls",
+				pendingBytes: -1,
+				flushLag:     -1,
+				replayLag:    -1,
+				writeLag:     -1,
+			},
 		}, nil)
 	} else {
 		table1 := "public.table1"

